File: src - java/Esercizi/CodaAPriorita/2/PriorityQueue.java
import Coda.Coda;

public class PriorityQueue {
	Coda coda_l;
	Coda coda_h;
	
	public PriorityQueue() {
		coda_l = new Coda();
		coda_h = new Coda();
	}

	public void pushLow(int el) {
		coda_l.push(el);
	}

	public void pushHigh(int el) {
		coda_h.push(el);
	}

	public int pop() {
		if (!coda_h.empty())
			return coda_h.pop();
		else
			return coda_l.pop();
	}

	public void clear() {
		coda_h.svuota();
		coda_l.svuota();
	}

	public boolean empty() {
		return coda_h.empty() && coda_l.empty();
	}
}

File: src - c++/Esercizi/CodaAPriorita/2/priorityqueue.cpp
#include "priorityqueue.h"

void PriorityQueue::PushLow(const TElem& e) {
  coda_l.Push(e);
}

void PriorityQueue::PushHigh(const TElem& e) {
  coda_h.Push(e);
}

TElem PriorityQueue::Pop() {
  if (!coda_h.Empty())
    return coda_h.Pop();
  else  
    return coda_l.Pop();
}

void PriorityQueue::Clear() {
  coda_h.Svuota();
  coda_l.Svuota();
}

bool PriorityQueue::Empty() const {
  return (coda_h.Empty() && coda_l.Empty());
}

